### DISCUSSION TITLE
Error Navigation cancelled from "xxx" to "xxx" with a new navigation.

### DISCUSSION THREAD

**Discussion by jackyuyuyu** [vuejs/router#1978] [Help and Questions] [answered]
请问这个报错是什么原因导致？怎么去解决
![image](https://github.com/vuejs/router/assets/144000915/cab722af-9af2-498f-8ab4-8860b97ea68a)
在以下这段源码中next()出现异常。
   return Promise.all(promises)
    }))
    // If not redirected
    if (!nextCalled) {
      **next()**
    }
  } catch (err) {
    const error = err || {}
    if (error.message === 'ERR_REDIRECT') {
      return this.$nuxt.$emit('routeChanged', to, from, error)

**Comment by posva** [comment, marked as answer]
This repository is Vue Router 4 where Navigation Failures were redesigned to avoid errors being thrown.

You can read more about this in docs and issues like https://github.com/vuejs/vue-router/issues/2881#issuecomment-520554378

- https://v3.router.vuejs.org/guide/advanced/navigation-failures.html
- https://router.vuejs.org/guide/advanced/navigation-failures.html#Waiting-for-the-result-of-a-Navigation

**Reply by jackyuyuyu** [reply]
是推荐使用vue-router 4吗？但是我的项目是nuxt构建的应用，nuxt的版本是2.15.8，现在无法升级nuxt版本，有没有其他方法解决这个报错呢
